src/bootstrap/compile.rs

```diff
@@ -640,6 +640,142 @@ impl Step for RustcLink {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct CodegenBackend {
+    pub target: TargetSelection,
+    pub compiler: Compiler,
+    pub backend: Interned<String>,
+}
+
+impl Step for CodegenBackend {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.paths(&["compiler/rustc_codegen_cranelift", "rustc_codegen_cranelift"])
```

Comment:
Is there a reason to double up here?

Reply:
I thought I had seen it somewhere that both the crate name and crate dir were allowed when using `.all_krates`, but I can't find it anymore. Will change it.